### PR TITLE
Remove state / isError and collect exception stacktrace in error logs

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>. NO
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderInterceptor.java [hunk 1]
```diff
@@ -92,12 +93,16 @@ private LogData transform(LoggingEvent event) {
...
         return -1 == ContextManager.getSpanId() ? builder.build()
                 : builder.setTraceContext(TraceContext.newBuilder()
                         .setTraceId(ContextManager.getGlobalTraceId())
                         .setSpanId(ContextManager.getSpanId())
                         .setTraceSegmentId(ContextManager.getSegmentId())
                         .build()).build();
     }
+
+    private String transformLogText(final LoggingEvent event) {
+        return event.getMessage() + "\n" + ThrowableTransformer.INSTANCE.convert2String(event.getThrowable(), 2048);
```
Comment (kezhenxu94) [inline apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderInterceptor.java:106]: I'm wondering whether this `2048` deserves a new config item, or the default value is suitable in this case @wu-sheng .

Comment (wu-sheng) [reply apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderInterceptor.java:106]: Let's keep this as default only. After all, in the next 8.5.0, I am expecting to use the file scraper to read the logs.

**Comment by kezhenxu94** [comment]
Screenshot would be 
image

**Review by wu-sheng — COMMENTED** [review]
I think you should delete `LogState` and related codes.

**Review by arugal — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
I think you forget to fix the e2e request codes
```
2021-01-29T00:40:36.7641667Z 00:40:36.760 INFO  o.a.skywalking.e2e.SimpleQueryClient.logs (414) - Query: {    "query": "query queryLogs($condition: LogQueryCondition) {        logs: queryLogs(condition: $condition) {            data: logs {                serviceName,                serviceId,                serviceInstanceName,                serviceInstanceId,                endpointName,                endpointId,                traceId,                timestamp,                isError,                contentType,                content,                tags{                    key, value                }            }        }    }",    "variables": {        "condition": {            "state": "ALL",            "serviceId": "WW91cl9BcHBsaWNhdGlvbk5hbWU=.1",            "endpointId": "",            "endpointName": "",            "queryDuration": {                "start": "2021-01-29 003821",                "end": "2021-01-29 004036",                "step": "SECOND"            },            "keywordsOfContent": [                            ],            "excludingKeywordsOfContent": [                            ],            "tags": [                {                    "key": "logger",                    "value": "org.apache.skywalking.e2e.controller.LogController"                }            ],            "paging": {                "pageNum": 1,                "pageSize": 15,                "needTotal": true            }        }    }}
2021-01-29T00:40:36.7879824Z 00:40:36.786 INFO  o.t.c.DockerComposeContainer.accept (64) - [/518476314fgsjzv_oap_1] STDOUT: 2021-01-29 00:40:36,785 - graphql.GraphQL -75499 [qtp831656273-40] WARN  [] - Query failed to validate : 'query queryLogs($condition: LogQueryCondition) {        logs: queryLogs(condition: $condition) {            data: logs {                serviceName,                serviceId,                serviceInstanceName,                serviceInstanceId,                endpointName,                endpointId,                traceId,                timestamp,                isError,                contentType,                content,                tags{                    key, value                }            }        }    }'
```

**isError** is still there.

**Review by wu-sheng — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]

**Review by wu-sheng — DISMISSED** [review]

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/Log.java [hunk 2]
```diff
@@ -34,7 +34,6 @@
     private String endpointName;
     private String traceId;
     private String timestamp;
-    private boolean isError;
     private String statusCode;
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/Log.java:37]: This should be deleted.

Comment (kezhenxu94) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/Log.java]: 🤦🏻‍♂️

**Review by wu-sheng — DISMISSED** [review]

**Review by wu-sheng — DISMISSED** [review]
LGTM

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**